utils/economic-checks/src/util.rs

```diff
@@ -0,0 +1,342 @@
+        TransactionPayment::on_finalize(current_blk);
+        Balances::on_finalize(current_blk);
+        System::on_finalize(current_blk);
+
+        let new_block_number = current_blk + 1;
+        System::set_block_number(new_block_number);
+        System::initialize(&new_block_number, &System::parent_hash(), &pre_digest);
+
+        // Run on_initialize hooks in order as they appear in AllPalletsWithSystem
+        System::on_initialize(new_block_number);
+        Balances::on_initialize(new_block_number);
+        TransactionPayment::on_initialize(new_block_number);
+        GearProgram::on_initialize(new_block_number);
+        GearMessenger::on_initialize(new_block_number);
+        Gear::on_initialize(new_block_number);
+        Usage::on_initialize(new_block_number);
+        Gas::on_initialize(new_block_number);
+        GearPayment::on_finalize(new_block_number);
+    }
+}
+
+pub(crate) fn run_to_block_with_ocw(
+    n: u32,
+    pool: Arc<RwLock<PoolState>>,
+    remaining_weight: Option<u64>,
+) {
+    // All blocks are to be authored by validator at index 0
+    let slot = Slot::from(0);
+    let pre_digest = Digest {
+        logs: vec![DigestItem::PreRuntime(AURA_ENGINE_ID, slot.encode())],
+    };
+
+    let now = System::block_number();
+    for i in now..n {
+        // Processing extrinsics in current block
+        process_tx_pool(pool.clone());
+        log::debug!("✅ Done processing transaction pool at block {}", i);
+        let remaining_weight =
+            remaining_weight.unwrap_or_else(<Runtime as pallet_gas::Config>::BlockGasLimit::get);
+
+        // Processing message queue
+        Gear::on_idle(i, remaining_weight);
+
+        // Offchain worker to charge rent
+        increase_offchain_time(1_000);
+        Usage::offchain_worker(i);
+
+        // on_finalize hooks (in pallets reverse order, as they appear in AllPalletsWithSystem)
+        GearPayment::on_finalize(i);
+        Gas::on_finalize(i);
+        Usage::on_finalize(i);
+        Gear::on_finalize(i);
+        GearMessenger::on_finalize(i);
+        GearProgram::on_finalize(i);
+        TransactionPayment::on_finalize(i);
+        Balances::on_finalize(i);
+        System::on_finalize(i);
+
+        let new_blk = i + 1;
+        System::set_block_number(i + 1);
+        log::debug!("📦 Initializing block {}", new_blk);
+        System::initialize(&new_blk, &System::parent_hash(), &pre_digest);
+
+        // Run on_initialize hooks in order as they appear in AllPalletsWithSystem
+        System::on_initialize(new_blk);
+        Gear::on_initialize(new_blk);
+        TransactionPayment::on_initialize(new_blk);
+        GearProgram::on_initialize(new_blk);
+        GearMessenger::on_initialize(new_blk);
+        Gear::on_initialize(new_blk);
+        Usage::on_initialize(new_blk);
+        Gas::on_initialize(new_blk);
+        GearPayment::on_finalize(new_blk);
+    }
+}
+
+fn increase_offchain_time(ms: u64) {
```

Comment:
increasing off chain time only seems not enough logic to me
I would like to increase it chained with running to some specific block

Reply:
well, it may not be necessary here it all - it's just an artifact from copy-and-paste the offchain workers related stuff from the `usage` pallet where offchain time did make difference for testing block_and_time locks in offchain workers. I think I'll go ahead removing this function completely

Reply:
I think this should be used, until we have ocw running. Its affection should hold for fuzz tests

Reply:
Leaving as is for the meantime - we can revisit it any time